usr/src/uts/common/fs/zfs/dmu_zfetch.c

```diff
@@ -197,9 +200,11 @@ dmu_zfetch_stream_create(zfetch_t *zf, uint64_t blkid)
  * routines to create, delete, find, or operate upon prefetch streams.
  */
 void
-dmu_zfetch(zfetch_t *zf, uint64_t blkid, uint64_t nblks)
+dmu_zfetch(zfetch_t *zf, uint64_t blkid, uint64_t nblks, boolean_t data)
```

Comment:
Can you describe the 'data' parameter in the comment above?  Seems like indirect blocks will always be prefetched, and 'data' controls if data blocks are also prefetched.  However, I think it would be even better to make the code more self-describing by renaming the parameter to something like "indirect_only" (and its sense reversed).